dask/dataframe/core.py

```diff
@@ -4596,6 +4596,13 @@ def set_index(
 
         if divisions is not None:
             check_divisions(divisions)
+        # mismatched sizes of indivual partitions = user error. how to fail gracefully?
+        elif (
+            getattr(other, "known_divisions", False)
+            and other.npartitions == self.npartitions
+        ):
+            pre_sorted = True
+            divisions = other.divisions
```

Comment:
@gjoseph92 I'm starting to question the premise of this section. 

This is causing other failures in `test_shuffle.py::test_set_index_general` when a series from the existing dataframe is used. The divisions are known and the `npartitions` match, so it does through this pass and ends up sorting the whole dataframe.

What kind of behavior would you like to see here?